Source/CsDebugScript.Engine/Engine/Debuggers/DbgEngDllHelpers/DebuggeFlowController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using DbgEngManaged;
+using System;
+
+namespace CsDebugScript.Engine.Debuggers.DbgEngDllHelpers
+{
+    /// <summary>
+    /// Controler for Debugee actions during live debugging.
+    /// </summary>
+    class DebuggeeFlowController
+    {
+        /// <summary>
+        /// Signal fired during interactive process debugging when debugee is released.
+        /// </summary>
+        public System.Threading.AutoResetEvent DebugStatusGo { get; private set; }
+
+        /// <summary>
+        /// Signal fired during interactive process debugging when debugee is interrupted.
+        /// </summary>
+        public System.Threading.AutoResetEvent DebugStatusBreak { get; private set; }
+
+        /// <summary>
+        /// Loop responsible for catching debug events and signaling debugee state.
+        /// </summary>
+        private System.Threading.Thread debuggerStateLoop;
+
+        /// <summary>
+        /// Syncronization signaling that debug callbacks are installed.
+        /// </summary>
+        private static readonly object eventCallbacksReady = new Object();
+
+        /// <summary>
+        /// Debug client for gbgeng interaction.
+        /// </summary>
+        private static IDebugClient client;
+
+        /// <summary>
+        /// Constructor.
```

Comment:
Done.

---
In reply to: [69981756](https://github.com/southpolenator/WinDbgCs/pull/8#discussion_r69981756) [](ancestors = 69981756)